File: web-front/src/main/java/com/isystk/sample/web/front/controller/api/v1/member/post/MemberPostRestController.java
package com.isystk.sample.web.front.controller.api.v1.member.post;

import static com.isystk.sample.common.Const.MESSAGE_SUCCESS;
import static com.isystk.sample.common.FrontUrl.API_V1_MEMBER_POSTS;

import com.isystk.sample.common.dto.Page;
import com.isystk.sample.common.helper.UserHelper;
import com.isystk.sample.common.util.ObjectMapperUtils;
import com.isystk.sample.domain.dto.TPostCriteria;
import com.isystk.sample.web.base.controller.api.AbstractRestController;
import com.isystk.sample.web.base.controller.api.resource.PageableResource;
import com.isystk.sample.web.base.controller.api.resource.PageableResourceImpl;
import com.isystk.sample.web.front.dto.FrontPostDto;
import com.isystk.sample.web.front.service.PostService;
import org.slf4j.Logger;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.ui.Model;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RestController;

@RestController
@RequestMapping(API_V1_MEMBER_POSTS)
public class MemberPostRestController extends AbstractRestController {

  private static final Logger log = org.slf4j.LoggerFactory
      .getLogger(MemberPostRestController.class);
  @Autowired
  PostService postService;

  @Autowired
  UserHelper userHelper;

  @Override
  public String getFunctionName() {
    return "API_MEMBER_POSTS";
  }

  /**
   * 会員表示
   *
   * @param query
   * @param model
   * @return
   */
  @GetMapping()
  public PageableResource index(MemberPostRestForm query, Model model) {
    // 入力値を詰め替える
    TPostCriteria criteria = new TPostCriteria();
    criteria.setUserIdEq(userHelper.getLoginUserId());
    criteria.setDeleteFlgFalse(true);
    criteria.setOrderBy("order by update_time desc");

    // 10件区切りで取得する
    Page<FrontPostDto> posts = postService.findAll(criteria, query);

    PageableResource resource = ObjectMapperUtils.map(posts, PageableResourceImpl.class);
    resource.setMessage(getMessage(MESSAGE_SUCCESS));

    return resource;
  }

}

File: web-front/src/main/kotlin/com/isystk/sample/web/front/controller/api/v1/member/post/MemberPostRestController.kt
package com.isystk.sample.web.front.controller.api.v1.member.post

import com.isystk.sample.common.Const
import com.isystk.sample.common.FrontUrl
import com.isystk.sample.common.helper.UserHelper
import com.isystk.sample.common.util.ObjectMapperUtils
import com.isystk.sample.domain.dto.TPostCriteria
import com.isystk.sample.web.base.controller.api.AbstractRestController
import com.isystk.sample.web.base.controller.api.resource.PageableResource
import com.isystk.sample.web.base.controller.api.resource.PageableResourceImpl
import com.isystk.sample.web.front.service.PostService
import org.slf4j.LoggerFactory
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.ui.Model
import org.springframework.web.bind.annotation.GetMapping
import org.springframework.web.bind.annotation.RequestMapping
import org.springframework.web.bind.annotation.RestController

@RestController
@RequestMapping(FrontUrl.API_V1_MEMBER_POSTS)
class MemberPostRestController : AbstractRestController() {
    @Autowired
    var postService: PostService? = null

    @Autowired
    var userHelper: UserHelper? = null
    override fun getFunctionName(): String {
        return "API_MEMBER_POSTS"
    }

    /**
     * 会員表示
     *
     * @param query
     * @param model
     * @return
     */
    @GetMapping
    fun index(query: MemberPostRestForm?, model: Model?): PageableResource {
        // 入力値を詰め替える
        val criteria = TPostCriteria()
        criteria.userIdEq = userHelper!!.loginUserId
        criteria.isDeleteFlgFalse = true
        criteria.orderBy = "order by update_time desc"

        // 10件区切りで取得する
        val posts = postService!!.findAll(criteria, query)
        val resource: PageableResource = ObjectMapperUtils.map(posts, PageableResourceImpl::class.java)
        resource.message = getMessage(Const.MESSAGE_SUCCESS)
        return resource
    }

    companion object {
        private val log = LoggerFactory
                .getLogger(MemberPostRestController::class.java)
    }
}